### PR TITLE
issue #31878 - xtupleapi.dtd updates

### PR DESCRIPTION
**Pull request by szuke**
Added 		ediinfo, glaccount, invoice, invoiceline, item and associated attributes

### CODE REVIEW

File: share/xml/xtupleapi.dtd [hunk 1]
```diff
@@ -347,9 +353,181 @@
 			ignore	(true|false)	'false'
 			key	CDATA	#FIXED	'customer_number, shipto_number' >
 
-<!ELEMENT custtax (customer_number, tax_authority, registration_number) >
+<!ELEMENT custtax (customer_number, tax_zone, registration_number) >
```
Comment (gilmoskowitz) [inline share/xml/xtupleapi.dtd]: The [custtax view definition](https://github.com/xtuple/xtuple/blob/v4.11.2/foundation-database/api/views/custtax.sql#L8) uses `tax_authority`.

File: share/xml/xtupleapi.dtd [hunk 2]
```diff
@@ -347,9 +353,181 @@
...
 <!ATTLIST custtax	ignore 	(true|false)	'false' >
 
+<!ELEMENT ediinfo (batchnumber, txnid, order_number, cust_po_number) >
+<!ATTLIST ediinfo	ignore 	(true|false)	'false' >
```
Comment (gilmoskowitz) [inline share/xml/xtupleapi.dtd]: I can't find an `ediinfo` view in `connect`, `qt-client`, `private-extensions`, or `xtuple`. Is this to support Core's EDI work?

File: share/xml/xtupleapi.dtd [hunk 3]
```diff
@@ -396,6 +574,7 @@
 		     misc_account_number,
 		     misc_charge)?,
 		   freight?,
+		   calculate_freight?,
```
Comment (gilmoskowitz) [inline share/xml/xtupleapi.dtd]: The [view](https://github.com/xtuple/xtuple/blob/v4.11.2/foundation-database/api/views/quote.sql#L50) doesn't have a `calculate_freight` column.

File: share/xml/xtupleapi.dtd [hunk 4]
```diff
@@ -630,6 +813,7 @@
 <!ELEMENT first			(#PCDATA) >
 <!ELEMENT fob			(#PCDATA) >
 <!ELEMENT freight		(#PCDATA) > <!-- Numeric -->
+<!ELEMENT calcuate_freight		(#PCDATA) > <!-- boolean -->
```
Comment (gilmoskowitz) [inline share/xml/xtupleapi.dtd]: typo - missing `l`

File: share/xml/xtupleapi.dtd [hunk 5]
```diff
@@ -711,16 +896,17 @@
 <!ELEMENT street_address2	(#PCDATA) >
 <!ELEMENT substitute_for 	(#PCDATA) >
 <!ELEMENT supplying_whs 	(#PCDATA) >
-<!ELEMENT tax_authority		(#PCDATA) >
```
Comment (gilmoskowitz) [inline share/xml/xtupleapi.dtd]: We still need `tax_authority` for the `custtax` and `customertaxreg` views.

File: share/xml/xtupleapi.dtd [hunk 6]
```diff
@@ -891,7 +1078,7 @@
 <!ATTLIST street_address2	quote	(true|false)	'true' >
 <!ATTLIST substitute_for 	quote	(true|false)	'true' >
 <!ATTLIST supplying_whs 	quote	(true|false)	'true' >
-<!ATTLIST tax_authority		quote	(true|false)	'true' >
+<!ATTLIST tax_zone		quote	(true|false)	'true' >
```
Comment (gilmoskowitz) [inline share/xml/xtupleapi.dtd]: We still need `tax_authority`'s attributes, too.

**Comment by szuke** [comment]
Ugh....sorry to put you through that testing. I knew there were some outstanding issues in that dtd. The ediinfo pieces are indeed to support the Core EDI stuff, but I'd like to commit those to the base. I'll remove those until we can vet the rest of it. 

I'll cleanup the attributes, etc.

Thanks...Scott

**Comment by gilmoskowitz** [comment]
Scott,

No problem - leave the EDI stuff there but add a comment in the DTD explaining why it's there.

**Comment by gilmoskowitz** [comment]
replaced by xtuple/qt-client#2084

**Pull request closed without merging**